src/requestforattestation/RequestForAttestation.ts

```diff
@@ -61,6 +63,95 @@ function getHashRoot(leaves: Uint8Array[]): Uint8Array {
   return hash(result)
 }
 
+function compressNonceAndHash(nonceHash: any[]): any {
+  if (nonceHash.length === 0) {
+    return {
+      hash: nonceHash[0],
+    }
+  }
+  return {
+    nonce: nonceHash[0],
+    hash: nonceHash[1],
+  }
+}
+
+export function compressClaimHashTree(
+  rfa: IRequestForAttestation
+): IRequestForAttestation['claimHashTree'] {
+  const result = {}
+  const claimTree = rfa.claimHashTree
+
+  Object.keys(claimTree).forEach(entryKey => {
+    result[entryKey] = Object.values(claimTree[entryKey])
+  })
+  return result
+}
+
+export function decompressClaimHashTree(
+  rfa: IRequestForAttestation[]
+): IRequestForAttestation['claimHashTree'] {
+  const result = {}
+
+  Object.keys(rfa).forEach(entryKey => {
+    result[entryKey] = compressNonceAndHash(Object.values(rfa[entryKey]))
```

Comment:
Suggestion:
`Object.values(rfa[entryKey])` in this function and `Object.values(claimTree[entryKey])` in the previous one are doing the same things. 
Maybe implement a helper function? Once you do so, you might also want to push it further, since you'll see that both this function and the previous one are very similar. You'll decide on the fly :)

Reply:
Timo was not so keen on the `compressNonceAndHash` helper function. Since we only use this once or twice I am not sure.

But we can bring this out.

Reply:
No problem 🙂